### ISSUE TITLE
False positive D402 when mentioning a different function that has the function name as suffix

### ISSUE THREAD

**Issue by Daraan** [astral-sh/ruff#13242]
I stumbled on the following false positive cases where the mention of `prefix_foo` flags a D402 on a function called `foo`

```python
def foo(self):
    """ Use prefix_foo() """
    # ^^^^^^^^^^^^^^^^^^^ D402
```

---

Version: `ruff 0.6.3`

Command: `ruff check . --select D402 --isolated`

Related #3769; where the same function is mentioned explicitly without a prefix.

**Comment by dhruvmanila** [comment]
I think we can solve this by using whole words instead, so look for `foo` and not any words that ends with `foo`. Curious to know what the ecosystem changes would look like.
